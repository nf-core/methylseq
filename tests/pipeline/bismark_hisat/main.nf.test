nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "bismark_hisat"
    tag "pipeline"

    test("Bismark Hisat Single-End") {
        when {
            params {
                aligner        = "bismark_hisat"
                save_reference = true
                outdir         = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/bismark_hisat/deduplicated/Ecoli_1_val_1_bismark_hisat2_pe.deduplicated.bam"),
                            path("$outputDir/bismark_hisat/deduplicated/SRR389222_trimmed_bismark_hisat2.deduplicated.bam"),
                            path("$outputDir/bismark_hisat/methylation_calls/bedGraph/Ecoli_1_val_1_bismark_hisat2_pe.deduplicated.bedGraph.gz"),
                            path("$outputDir/bismark_hisat/methylation_calls/bedGraph/SRR389222_trimmed_bismark_hisat2.deduplicated.bedGraph.gz"),
                            path("$outputDir/bismark_hisat/methylation_calls/mbias/Ecoli_1_val_1_bismark_hisat2_pe.deduplicated.M-bias.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/mbias/SRR389222_trimmed_bismark_hisat2.deduplicated.M-bias.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/methylation_calls/").list(),
                            path("$outputDir/bismark_hisat/methylation_calls/methylation_coverage/Ecoli_1_val_1_bismark_hisat2_pe.deduplicated.bismark.cov.gz"),
                            path("$outputDir/bismark_hisat/methylation_calls/methylation_coverage/SRR389222_trimmed_bismark_hisat2.deduplicated.bismark.cov.gz"),
                            path("$outputDir/bismark_hisat/methylation_calls/splitting_report/Ecoli_1_val_1_bismark_hisat2_pe.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/splitting_report/SRR389222_trimmed_bismark_hisat2.deduplicated_splitting_report.txt"),
                            path("$outputDir/cat/SRR389222.merged.fastq.gz"),
                            path("$outputDir/qualimap/Ecoli/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222/genome_results.txt")).match("single_end") },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_end_software_versions") },
                { assert new File("$outputDir/bismark_hisat/alignments/logs/Ecoli_1_val_1_bismark_hisat2_PE_report.txt").exists() },
                { assert new File("$outputDir/bismark_hisat/alignments/logs/SRR389222_trimmed_bismark_hisat2_SE_report.txt").exists() },
                { assert new File("$outputDir/bismark_hisat/reports/Ecoli_1_val_1_bismark_hisat2_PE_report.html").exists() },
                { assert new File("$outputDir/bismark_hisat/reports/SRR389222_trimmed_bismark_hisat2_SE_report.html").exists() },
                { assert new File("$outputDir/bismark_hisat/summary/bismark_summary_report.html").exists() },
                { assert new File("$outputDir/bismark_hisat/summary/bismark_summary_report.txt").exists() },
                { assert new File("$outputDir/fastqc/Ecoli_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/Ecoli_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_fastqc.html").exists() },
                { assert new File("$outputDir/multiqc/bismark_hisat/multiqc_report.html").exists() },
                { assert new File("$outputDir/trimgalore/logs/Ecoli_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/logs/Ecoli_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/logs/SRR389222.fastq.gz_trimming_report.txt").exists() }
            )
        }
    }

    test("Bismark Hisat Single-End With Index") {
        when {
            params {
                aligner         = "bismark_hisat"
                bismark_index   = "https://github.com/nf-core/test-datasets/raw/methylseq/reference/Hisat2_Index.tar.gz"
                local_alignment = true
                outdir          = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
            )
        }
    }

    test("Bismark Hisat Single-End With RRBS") {
        when {
            params {
                aligner        = "bismark_hisat"
                save_reference = true
                skip_trimming  = true
                rrbs           = true
                outdir         = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/bismark_hisat/alignments/Ecoli_10K_methylated_R1_bismark_hisat2_pe.bam"),
                            path("$outputDir/bismark_hisat/alignments/SRR389222.merged_bismark_hisat2.bam"),
                            path("$outputDir/bismark_hisat/methylation_calls/bedGraph/Ecoli_10K_methylated_R1_bismark_hisat2_pe.bedGraph.gz"),
                            path("$outputDir/bismark_hisat/methylation_calls/bedGraph/SRR389222.merged_bismark_hisat2.bedGraph.gz"),
                            path("$outputDir/bismark_hisat/methylation_calls/mbias/Ecoli_10K_methylated_R1_bismark_hisat2_pe.M-bias.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/mbias/SRR389222.merged_bismark_hisat2.M-bias.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/methylation_calls/").list(),
                            path("$outputDir/bismark_hisat/methylation_calls/methylation_coverage/Ecoli_10K_methylated_R1_bismark_hisat2_pe.bismark.cov.gz"),
                            path("$outputDir/bismark_hisat/methylation_calls/methylation_coverage/SRR389222.merged_bismark_hisat2.bismark.cov.gz"),
                            path("$outputDir/bismark_hisat/methylation_calls/splitting_report/Ecoli_10K_methylated_R1_bismark_hisat2_pe_splitting_report.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/splitting_report/SRR389222.merged_bismark_hisat2_splitting_report.txt"),
                            path("$outputDir/cat/SRR389222.merged.fastq.gz"),
                            path("$outputDir/qualimap/Ecoli/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222/genome_results.txt")).match("single_end_rrbs") },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_end_rrbs_software_versions") },
                { assert new File("$outputDir/bismark_hisat/alignments/logs/Ecoli_10K_methylated_R1_bismark_hisat2_PE_report.txt").exists() },
                { assert new File("$outputDir/bismark_hisat/alignments/logs/SRR389222.merged_bismark_hisat2_SE_report.txt").exists() },
                { assert new File("$outputDir/bismark_hisat/reports/Ecoli_10K_methylated_R1_bismark_hisat2_PE_report.html").exists() },
                { assert new File("$outputDir/bismark_hisat/reports/SRR389222.merged_bismark_hisat2_SE_report.html").exists() },
                { assert new File("$outputDir/bismark_hisat/summary/bismark_summary_report.html").exists() },
                { assert new File("$outputDir/bismark_hisat/summary/bismark_summary_report.txt").exists() },
                { assert new File("$outputDir/fastqc/Ecoli_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/Ecoli_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_fastqc.html").exists() },
                { assert new File("$outputDir/multiqc/bismark_hisat/multiqc_report.html").exists() }
            )
        }
    }

    test("Bismark Hisat Single-End With RRBS With Index") {
        when {
            params {
                aligner         = "bismark_hisat"
                skip_trimming   = true
                rrbs            = true
                bismark_index   = "https://github.com/nf-core/test-datasets/raw/methylseq/reference/Hisat2_Index.tar.gz"
                local_alignment = true
                outdir          = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
            )
        }
    }

}
