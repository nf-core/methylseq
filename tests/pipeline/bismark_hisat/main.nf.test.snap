{
    "single_end_software_versions": {
        "content": [
            "{BISMARK_ALIGN={bismark=0.24.0}, BISMARK_GENOMEPREPARATION={bismark=0.24.0}, CAT_FASTQ={cat=8.30}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, FASTQC={fastqc=0.11.9}, QUALIMAP_BAMQC={qualimap=2.2.2-dev}, SAMPLESHEET_CHECK={python=3.8.3}, TRIMGALORE={cutadapt=3.4, trimgalore=0.6.7}, Workflow={nf-core/methylseq=2.4.0}}"
        ],
        "timestamp": "2023-06-01T02:43:30+0000"
    },
    "single_end_rrbs_software_versions": {
        "content": [
            "{BISMARK_ALIGN={bismark=0.24.0}, BISMARK_GENOMEPREPARATION={bismark=0.24.0}, CAT_FASTQ={cat=8.30}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, FASTQC={fastqc=0.11.9}, QUALIMAP_BAMQC={qualimap=2.2.2-dev}, SAMPLESHEET_CHECK={python=3.8.3}, Workflow={nf-core/methylseq=2.4.0}}"
        ],
        "timestamp": "2023-06-01T02:43:30+0000"
    },
    "single_end": {
        "content": [
            "Ecoli_10K_methylated_1_val_1_bismark_hisat2_pe.deduplicated_splitting_report.txt:md5,bdec8746a4f29c56ddbe37a92f7d35dd",
            "SRR389222_sub1_trimmed_bismark_hisat2.deduplicated_splitting_report.txt:md5,4020063dd8610b58b7b39c3ca62cef17",
            "SRR389222_sub2_trimmed_bismark_hisat2.deduplicated_splitting_report.txt:md5,4db2155fdfae1cb9c7c758edba256e2e",
            "genome_results.txt:md5,cd5d3b03821c038eb3bda24beb3ae938",
            "genome_results.txt:md5,9956ed1dcf51e6975f4a6ab8fe156aba",
            "genome_results.txt:md5,1133ea5d1376389c6a17def89d2e90ec"
        ],
        "timestamp": "2023-06-01T02:43:30+0000"
    },
    "single_end_rrbs": {
        "content": [
            "Ecoli_10K_methylated_R1_bismark_hisat2_pe_splitting_report.txt:md5,76868c2015864a3f7c2b4cd1738e4765",
            "SRR389222_sub1_bismark_hisat2_splitting_report.txt:md5,754ed7bf4d75acee466c48c05a11d331",
            "SRR389222_sub2.merged_bismark_hisat2_splitting_report.txt:md5,0629e1f8b4727efe8fd69500783eec61",
            "genome_results.txt:md5,3afd1eb7d4dc72a05e4d4de887d791e3",
            "genome_results.txt:md5,58a2aeeb5eacca4c766bb23fee8f9013",
            "genome_results.txt:md5,473022a29cf12d97e75451a4eea67c02"
        ],
        "timestamp": "2023-06-01T02:43:30+0000"
    }
}